runtime/parser2/lexer/lexer.go

```diff
@@ -229,6 +242,11 @@ func (l *lexer) acceptOne(r rune) bool {
 
 // emit writes a token to the channel.
 func (l *lexer) emit(ty TokenType, val interface{}, rangeStart ast.Position, consume bool) {
+
+	if len(l.tokens) >= tokenLimit {
+		panic(TokenLimitReachedError{})
```

Comment:
Should this be wrapped in `FatalError`? I'm not sure of the flow of control so I really don't know.

Reply:
Tokenization happens before parsing, so there is no real need to wrap it. We might want to clean up / unify the error handling in general